surf-auth/surfAuthDebot.sol

```diff
@@ -0,0 +1,128 @@
+pragma ton-solidity >=0.35.0;
+pragma AbiHeader expire;
+pragma AbiHeader time;
+pragma AbiHeader pubkey;
+
+import "../Debot.sol";
+import "../Terminal.sol";
+import "../Menu.sol";
+import "../AddressInput.sol";
+import "../ConfirmInput.sol";
+import "../Upgradable.sol";
+import "../Sdk.sol";
+import "./Network.sol";
+import "./Base64.sol";
+import "./SigningBoxInput.sol";
+import "./UserInfo.sol";
+
+
+contract TodoDebot is Debot, Upgradable {
+    string m_id;
+    string m_pin;
+    string m_otp;
+    string m_callbackUrl;
+    uint32 m_sigingBoxHandle;
+    uint256 m_userPK;
+
+    bytes m_icon;
+
+    function start() public override { }
+
+    /// @notice Returns Metadata about DeBot.
+    function getDebotInfo() public functionID(0xDEB) override view returns(
+        string name, string version, string publisher, string key, string author,
+        address support, string hello, string language, string dabi, bytes icon
+    ) {
+        name = "Surf Auth DeBot";
+        version = "0.2.0";
+        publisher = "TON Labs";
+        key = "Auth";
+        author = "TON Labs";
+        support = address.makeAddrStd(0, 0x66e01d6df5a8d7677d9ab2daf7f258f1e2a7fe73da5320300395f99e01dc3b5f);
+        hello = "Hi, i'm a Surf Auth DeBot.";
+        language = "en";
+        dabi = m_debotAbi.get();
+        icon = m_icon;
+    }
+
+    function getRequiredInterfaces() public view override returns (uint256[] interfaces) {
+        return [ Terminal.ID, Menu.ID, AddressInput.ID, ConfirmInput.ID ];
```

Comment:
you use
Terminal UserInfo Sdk Base64 Network

Reply:
Done